### PR TITLE
Host Command improvements

### PR DESCRIPTION
**Pull request by niedzwiecki-dawid**
A few improvements of Host Command subsystem:
-handling IN_PROGRESS status
-common function for signaling a new function
-user callback

### CODE REVIEW

File: subsys/mgmt/ec_host_cmd/ec_host_cmd_handler.c [hunk 1]
```diff
@@ -161,6 +185,35 @@ int ec_host_cmd_send_response(enum ec_host_cmd_status status,
...
+		LOG_INF("HC pending done, size=%d, result=%d",
+			args->output_buf_size, status);
+
+		/* Don't support saving response data, so mark the response as unavailable
+		 * in that case.
+		 */
+		if (args->output_buf_size != 0)
+			saved_status = EC_HOST_CMD_UNAVAILABLE;
+		else
+			saved_status = status;
```
Comment (fabiobaltieri) [inline subsys/mgmt/ec_host_cmd/ec_host_cmd_handler.c:210]: add braces (surprised the CI did not catch it...)

File: include/zephyr/mgmt/ec_host_cmd/ec_host_cmd.h [hunk 2]
```diff
@@ -260,6 +265,14 @@ int ec_host_cmd_init(struct ec_host_cmd_backend *backend);
 int ec_host_cmd_send_response(enum ec_host_cmd_status status,
 		const struct ec_host_cmd_handler_args *args);
 
+/**
+ * @brief Signal a new host command
+ *
+ * Signal that a new host command has been received. The function should be called by a backend
+ * after copying data to the rx buffer and setting the length.
+ */
+void ec_host_cmd_received(void);
```
[the remark this answers is not in the document]
Comment (semihalf-niedzwiecki-dawid) [reply include/zephyr/mgmt/ec_host_cmd/ec_host_cmd.h]: You are right, I wanted to use kind of CrosEC terminology(`host_command_received`), but it doesn't make sense. What do you think about `ec_host_cmd_rx_notify`?

File: subsys/mgmt/ec_host_cmd/ec_host_cmd_handler.c (hunk 1 above)
Comment (semihalf-niedzwiecki-dawid) [reply subsys/mgmt/ec_host_cmd/ec_host_cmd_handler.c]: Done.

**Review by fabiobaltieri — DISMISSED** [review]

File: subsys/mgmt/ec_host_cmd/Kconfig [hunk 3]
```diff
@@ -87,6 +87,13 @@ config EC_HOST_CMD_DEDICATED_THREAD
 	  The ec_host_cmd_init function creates a new thread dedicated for Host Command.
 	  Otherwise the ec_host_cmd_task function has to be called within another thread.
 
+config EC_HOST_CMD_IN_PROGRESS_STATUS
+	bool "Save status of last IN_PROGRESS command"
```
Comment (keith-zephyr) [inline subsys/mgmt/ec_host_cmd/Kconfig:91]: Should this include `default y if EC_HOST_CMD_BACKEND_SHI`?

File: subsys/mgmt/ec_host_cmd/ec_host_cmd_handler.c [hunk 4]
```diff
@@ -51,6 +51,14 @@ static struct ec_host_cmd ec_host_cmd = {
 		},
 };
 
+#ifdef CONFIG_EC_HOST_CMD_IN_PROGRESS_STATUS
+/* Indicates that a command has sent EC_HOST_CMD_IN_PROGRESS but hasn't sent a final status */
+static uint8_t cmd_in_progress;
```
Comment (keith-zephyr) [inline subsys/mgmt/ec_host_cmd/ec_host_cmd_handler.c:56]: Please use a `bool`.

File: include/zephyr/mgmt/ec_host_cmd/ec_host_cmd.h [hunk 5]
```diff
@@ -282,6 +282,31 @@ const struct ec_host_cmd *ec_host_cmd_get_hc(void);
 FUNC_NORETURN void ec_host_cmd_task(void);
 #endif
 
+#ifdef CONFIG_EC_HOST_CMD_IN_PROGRESS_STATUS
```
Comment (keith-zephyr) [inline include/zephyr/mgmt/ec_host_cmd/ec_host_cmd.h:326]: Note - the commit message has a typo. Delete extra "lat"

"Add a config to save the final result of a host command that has"

File: subsys/mgmt/ec_host_cmd/ec_host_cmd_handler.c (hunk 4 above)
Comment (semihalf-niedzwiecki-dawid) [reply subsys/mgmt/ec_host_cmd/ec_host_cmd_handler.c]: Done.

File: include/zephyr/mgmt/ec_host_cmd/ec_host_cmd.h (hunk 5 above)
Comment (semihalf-niedzwiecki-dawid) [reply include/zephyr/mgmt/ec_host_cmd/ec_host_cmd.h:326]: Done.

File: subsys/mgmt/ec_host_cmd/Kconfig (hunk 3 above)
Comment (semihalf-niedzwiecki-dawid) [reply subsys/mgmt/ec_host_cmd/Kconfig:91]: Makes sense. Done.

**Review by fabiobaltieri — APPROVED** [review]

**Review by keith-zephyr — APPROVED** [review]

**Pull request merged**